src/test/java/spoon/test/ctType/CtTypeTest.java

```diff
@@ -84,27 +96,98 @@ public void testIsSubTypeOf() throws Exception {
 	public void testIsSubTypeOfonTypeParameters() throws Exception {
 		CtType<X> xCtType = buildClass(X.class);
+		Factory factory = xCtType.getFactory();
+
+		CtType<?> oCtType = factory.Type().get("spoon.test.ctType.testclasses.O");
+		CtType<?> pCtType = factory.Type().get("spoon.test.ctType.testclasses.P");
+		CtTypeReference<?> objectCtTypeRef = factory.Type().OBJECT;
+
+		List<CtTypeParameter> oTypeParameters = oCtType.getFormalCtTypeParameters();
+		assertTrue(oTypeParameters.size() == 1);
+		List<CtTypeParameter> pTypeParameters = pCtType.getFormalCtTypeParameters();
+		assertTrue(pTypeParameters.size() == 2);
+
+		CtType<?> O_A_CtType = oTypeParameters.get(0);
+		CtType<?> P_D_CtType = pTypeParameters.get(0);
+		CtType<?> P_F_CtType = pTypeParameters.get(1);
+
+		CtMethod<?> O_FooMethod = oCtType.filterChildren(new NameFilter<>("foo")).first();
+		CtMethod<?> P_FooMethod = pCtType.filterChildren(new NameFilter<>("foo")).first();
+
+		CtType<?> O_B_CtType = O_FooMethod.getType().getDeclaration();
+		CtType<?> P_E_CtType = P_FooMethod.getType().getDeclaration();
+
+		assertTrue(O_B_CtType.isSubtypeOf(xCtType.getReference()));
+		assertTrue(O_B_CtType.isSubtypeOf(O_A_CtType.getReference()));
+		
+		assertTrue(P_E_CtType.isSubtypeOf(xCtType.getReference()));
+		assertTrue(P_E_CtType.isSubtypeOf(P_D_CtType.getReference()));
+		assertTrue(P_E_CtType.isSubtypeOf(O_A_CtType.getReference()));
+		
+		assertTrue(P_D_CtType.isSubtypeOf(O_A_CtType.getReference()));
+		assertTrue(P_E_CtType.isSubtypeOf(O_B_CtType.getReference()));
+
+		assertTrue(P_E_CtType.isSubtypeOf(objectCtTypeRef));
+		assertTrue(P_F_CtType.isSubtypeOf(objectCtTypeRef));
+	}
+	
+	@Test
+	public void testIsSubTypeOfonTypeReferences() throws Exception {
+		final Launcher launcher = new Launcher();
+		launcher.setArgs(new String[]{"-c"});
+		launcher.addInputResource("./src/test/java/spoon/test/ctType/testclasses/SubtypeModel.java");
+		launcher.buildModel();
+		Factory factory = launcher.getFactory();
+		
+		CtType<?> oCtType = factory.Class().get("spoon.test.ctType.testclasses.SubtypeModel");
+		CtMethod<?> O_FooMethod = oCtType.filterChildren(new NameFilter<>("foo")).first();
+
+		Map<String, CtTypeReference<?>> nameToTypeRef = new HashMap<>();
+		O_FooMethod.filterChildren(new TypeFilter<>(CtLocalVariable.class)).forEach((CtLocalVariable var)->{
+			nameToTypeRef.put(var.getSimpleName(), var.getType());
+		});
+		
+		int[] count = new int[1];
+
+		O_FooMethod.filterChildren(new TypeFilter<>(CtAssignment.class)).forEach((CtAssignment ass)->{
+			for (CtComment comment : ass.getComments()) {
+				checkIsNotSubtype(comment, nameToTypeRef);
+				count[0]++;
+			};
+			count[0]++;
+			checkIsSubtype(((CtVariableAccess) ass.getAssigned()).getVariable().getType(), ((CtVariableAccess) ass.getAssignment()).getVariable().getType(), nameToTypeRef);
+		});
+		
+		assertTrue(count[0]>(9*8));
+	}
 
-		CtType<?> oCtType = xCtType.getFactory().Type().get("spoon.test.ctType.testclasses.O");
```

Comment:
Github comparer confused us. In the better comparer there is visible that red line 91 is not deleted by is located on green line 105, etc.
So I did not deleted test code ;)